### PR TITLE
Fix track edition and change http code on unknwon objet id

### LINKED ISSUE #4171: Edit a track section fails
### What happened?

When trying to edit any field of a track section the update fails.

The request payload looks like this:

```json
[
  {
    "operation_type": "UPDATE",
    "obj_id": "osrd/editor/new-signal-id", # <--- This is wrong it should be the current track id
    "obj_type": "TrackSection",
    "railjson_patch": [
      ...
    ]
  }
]
```

Note: Editoast shouldn't fail with a 500. We should be able to detect an invalid id. To do so we need to check the response of this query: https://github.com/DGEXSolutions/osrd/blob/c8a1770ca54769c13e51c60931707d364e1ec781/editoast/src/schema/operation/update.rs#L24-L30

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the editor
2. Select a track section
3. Edit your selection
4. Try to save the object
5. An error 500 happen

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

c8a1770ca54769c13e51c60931707d364e1ec781

### PR DESCRIPTION
**Pull request by Castavo**
closes: #4171

### CODE REVIEW

File: editoast/src/schema/operation/mod.rs [hunk 1]
```diff
@@ -58,9 +58,9 @@ impl Operation {
 #[editoast_error(base_id = "operation")]
 enum OperationError {
     // To modify
-    #[error("Object '{0}', could not be found")]
+    #[error("Object '{0}', could not be found in the infrastructure '{1}'")]
     #[editoast_error(status = 404)]
-    ObjectNotFound(String),
+    ObjectNotFound(String, i64),
```
Comment (flomonster) [inline editoast/src/schema/operation/mod.rs]: We could enhance the error payload.
```suggestion
    ObjectNotFound{obj_id: String, infra_id: i64},
```

**Review by flomonster — APPROVED** [review]
LGTM

**Review by leovalais — APPROVED** [review]

**Review by SharglutDev — APPROVED** [review]
Nice fix :) Successfully tested.
Just maybe in tools.tsx, fix the import not used and the indent warnings

**Review by clarani — APPROVED** [review]
I didn't tested in local, but @SharglutDev did and I corrected the bug with @Castavo

**Pull request merged**